pkgs/development/haskell-modules/configuration-common.nix

```
@@ -1440,6 +1440,21 @@ self: super: {
     HsYAML = self.HsYAML_0_2_1_0;
   };
 
+  # Needed for ghcide
+  haskell-lsp_0_19_0_0 = super.haskell-lsp_0_19_0_0.override {
+    haskell-lsp-types = super.haskell-lsp-types_0_19_0_0;
+  };
+
+  # this will probably need to get updated with every ghcide update,
+  # we need an override because ghcide is tracking haskell-lsp closely.
+  ghcide = dontCheck (super.ghcide.override rec {
+    haskell-lsp-types = super.haskell-lsp-types_0_19_0_0;
```

```suggestion
    haskell-lsp-types = self.haskell-lsp-types_0_19_0_0;
  };

  # this will probably need to get updated with every ghcide update,
  # we need an override because ghcide is tracking haskell-lsp closely.
  ghcide = dontCheck (super.ghcide.override rec {
    haskell-lsp-types = self.haskell-lsp-types_0_19_0_0;
```

In general, derivations that are being overriden need to come from `super`, and dependencies need to come from `self`.

---

Since there are no overrides for lsp-types in the file, I didn‘t  think it was necessary. But yes, it makes sense, when the package breaks and needs to overriden manually.